packages/chain-helpers/src/blockchain/Blockchain.utils.ts

```diff
@@ -116,48 +90,41 @@ export async function dispatchTx(
 
   const unsubscribe = await tx.send(subscription)
 
-  return promise.finally(() => unsubscribe())
+  return promise
+    .catch((e) => Promise.reject(ErrorHandler.getExtrinsicError(e) || e))
+    .finally(() => unsubscribe())
 }
+export const dispatchTx = submitSignedTx
 
 /**
- * Checks the TxError for relevant ones and returns these as matched SDKError for recoverability.
+ * Checks the TxError/TxStatus for issues that may be resolved via resigning.
  *
- * @param reason Polkadot API returned error.
- * @returns If matched, a SDKError, else original reason.
+ * @param reason Polkadot API returned error or ISubmittableResult.
+ * @returns Whether or not this issue may be resolved via resigning.
  */
-function matchTxError(reason: Error): SDKErrors.SDKError | Error {
-  switch (true) {
-    case reason.message.includes(TxOutdated):
-      return SDKErrors.ERROR_TRANSACTION_OUTDATED()
-    case reason.message.includes(TxPriority):
-      return SDKErrors.ERROR_TRANSACTION_PRIORITY()
-    case reason.message.includes(TxDuplicate):
-      return SDKErrors.ERROR_TRANSACTION_DUPLICATE()
-    default:
-      return reason
+export function isRecoverableTxError(
+  reason: Error | ISubmittableResult
+): boolean {
+  if (reason instanceof Error) {
+    return (
+      reason.message.includes(TxOutdated) ||
+      reason.message.includes(TxPriority) ||
+      reason.message.includes(TxDuplicate) ||
+      false
+    )
   }
+  if (
+    reason &&
+    typeof reason === 'object' &&
+    typeof reason.status === 'object'
+  ) {
+    const { status } = reason as ISubmittableResult
+    if (status.isUsurped) return true
+    // if (status.isRetracted) return true
```

Comment:
Is this still needed?

Reply:
no, this can go - retracted actually means the transaction goes back in the pool, so we would not want to resubmit. Unsure about `dropped` and `invalid` though

Reply:
I think it is ok to not regard them here.